warrior/Framework/Utils/encryption_utils.py

```diff
@@ -80,16 +82,21 @@ def encrypt(message, encoded_key=False):
 def decrypt(message, encoded_key=False):
     """This is decryption"""
     iv, cipher = get_key(encoded_key)
-    try:
-        # in python2, can just use message.decode("hex")
-        # but in python3, str.decode is removed
-        return str(cipher.decrypt(codecs.decode(message, "hex_codec"))[len(iv):], 'utf-8')
-    except binascii.Error as err:
-        # This is dangerous...
-        # using exception to handle if encrypted/not encrypted condition
-        return message
-    except Exception:
-        print_error(traceback.format_exc())
+    if cipher is None:
+        print_warning("encrypted messages if any, can't be decrypted")
+        decrypt_message = message
+    else:
+        try:
+            # in python2, can just use message.decode("hex")
+            # but in python3, str.decode is removed
+            decrypt_message = str(cipher.decrypt(codecs.decode(message, "hex_codec"))[len(iv):], 'utf-8')
+        except binascii.Error as err:
+            # This is dangerous...
+            # using exception to handle if encrypted/not encrypted condition
+            decrypt_message = message
+        except Exception:
+            print_error(traceback.format_exc())
```

Comment:
Do we need line 96 to be repeated after the traceback printing?

Reply:
yes. if message is not encrypted (plain text), it returns as it is with this piece of code